engines/osxnative/user.go

```diff
@@ -0,0 +1,149 @@
+package osxnative
+
+import (
+	"github.com/satori/go.uuid"
+	"os"
+	"path"
+	"strconv"
+)
+
+type user struct {
+	d    dscl
+	name string
+}
+
+func (u *user) create() error {
+	uid, err := u.getMaxUid()
+	if err != nil {
+		return err
+	}
+
+	staffGid, err := u.d.read("/Groups/staff", "PrimaryGroupID")
+	if err != nil {
+		u.delete()
+		return err
+	}
+
+	gid, err := strconv.Atoi(staffGid)
+	if err != nil {
+		return err
+	}
+
+	name := "worker-" + uuid.NewV4().String()
+	userPath := path.Join("/Users", name)
+
+	err = u.d.create(userPath)
+	if err != nil {
+		return err
+	}
+
+	// We set the uid, then check if it is unique, if
+	// not, increment and try again
+	duplicated := true
+	for duplicated {
+		uid++
+		strUid := strconv.FormatInt(int64(uid), 10)
+
+		err = u.d.create(userPath, "uid", strUid)
+		if err != nil {
+			u.delete()
+			return err
+		}
+
+		duplicated, err = u.isDuplicatedId("uid", uid)
+		if err != nil {
+			u.delete()
+			return err
+		}
+	}
+
+	err = u.d.create(userPath, "PrimaryGroupID", staffGid)
+	if err != nil {
+		u.delete()
+		return err
+	}
+
+	err = u.d.create(userPath, "NFSHomeDirectory", userPath)
+	if err != nil {
+		u.delete()
+		return err
+	}
+
+	err = os.MkdirAll(userPath, 0700)
+	if err != nil {
+		u.delete()
+		return err
+	}
+
+	err = os.Chown(userPath, uid, gid)
```

Comment:
hrm, would that allow the uid to have ownership and anyone within the admin group? If so, that might work.